### PR TITLE
Move defaults to globalSettings (fix #296), try #2

### LINKED ISSUE #296: `EclipseKeys.withSource in ThisBuild` is ignored (4.0.0 regression from 3.0.0)
I have a multi-project build, and set 
```
EclipseKeys.withSource in ThisBuild := true
```
in [the main build.sbt](https://github.com/inc-lc/ilc-scala/blob/master/build.sbt#L105). If I switch from 2.5.0 to 4.0.0, that stops working, because of the `in ThisBuild` part — somehow I need to set that per project. 3.0.0 is safe. Has that got anything to do with AutoPlugin?

### PR DESCRIPTION
**Pull request by Blaisorblade**
New edition of #299, bumping major version as requested by @jsuereth.

## Original description
[SBT docs](http://www.scala-sbt.org/0.13/docs/Plugins.html#projectSettings+and+buildSettings) recommend:
“Use globalSettings to define the default value of a setting.”

`sbteclipse` doesn't do that, causing #296 and other variants. This PR fixes that bug. However, I'm not 100% sure that I'm moving all settings that should be moved.

### CODE REVIEW

**Comment by Blaisorblade** [comment]
@jsuereth, ping? This is identical to the reviewed #299, modulo the version.

**Comment by Blaisorblade** [comment]
@jsuereth, @benmccann ping again? This is a reviewed PR sitting around for a few months (and still not conflicting) — a very low-hanging fruit for you! :smile:

**Comment by benmccann** [comment]
This seems reasonable, but I really don't know SBT. I have to defer to @jsuereth on that one since he's the SBT expert. I've been reviewing things with Eclipse-specific changes.

**Comment by Blaisorblade** [comment]
Let me try again. @benmccann, note that @jsuereth had already reviewed this (as #299); I only changed the major release as he suggested.

**Comment by benmccann** [comment]
@Blaisorblade can you rebase it so that there are no merge conflicts?

**Comment by Blaisorblade** [comment]
Done. The only conflict was in fact trivial (I replaced Blaisorblade@463880247155dfb09184c1b2c6f9d0b135f9cd03 with Blaisorblade@87004212d2da5807126c41a5473facdfe1e63ede).

**Comment by Blaisorblade** [comment]
Looking quickly into the failure. I think the issue is due to the interaction with #309; I've tried adding a fix (commit explains the rationale), let's see how this proceeds.

**Comment by Blaisorblade** [comment]
FYI: the fix worked as I expected. @benmccann would you want to merge this or do you want to ping somebody else? Maybe Josh is less busy now...

@jroper I updated my PR because of the interaction with your #309; your test passes, would you be willing to review my change (08171e2d4efb68ff59d1b4d2495eae88cbf5d2b7) to check it's still fine for you and makes sense?

**Comment by benmccann** [comment]
Thanks. I'll probably take a look, but I'm much stronger on the Eclipse side than the SBT side, so yeah I may track down Josh or SBT folks to review.

The other thing that should be cleaned up is to squash the six commits into a single commit:

    git rebase -i HEAD~6
    git push -f

**Comment by Blaisorblade** [comment]
Squashing done.

BTW, I've tried summarizing the PR well enough in the build summary (I had also tried earlier, hopefully now I did it better). I believe the needed SBT knowledge is in this paragraph, though I can understand your feeling:

> Build settings can be set at the global level, at the build level or at
> the project level, in order of increasing priority. To allow overriding
> settings for a whole build, SBT guidelines recommend that the defaults
> be set at the global level. This PR does that wherever possible.

**Comment by dwijnand** [comment]
I've not seen anyone argue against using `globalSettings` to set defaults, so in general I see no reason not to follow what the docs recommend.

**Comment by jroper** [comment]
The change looks fine with respect to my PR.

In general, I think moving anything that someone might want to configure globally (in `~/.sbt/0.13/global.sbt`) to global settings makes a lot of sense - it's my opinion that IDE plugins should not be included in projects but should be configured by a user globally, and so allowing the user to manage that configuration themselves is sensible.

**Comment by benmccann** [comment]
@Blaisorblade thank you for sticking with this and handling it very professionally

I can't publish SBT plugins unfortunately. Josh tried to give me access awhile back and there was some problem I never got fully debugged, so you'll have to get one of the Lightbend folks to cut a new release to be able to use this.
